api/v1alpha1/database_webhook.go

```diff
@@ -188,10 +193,35 @@ func instanceSpecValidatorForCreate(instance *Instance, allErrs field.ErrorList,
 		))
 	}
 
+	if isInvalidAdditionalArguments, allowedAdditionalArguments := isAdditionalArgumentsInvalid(instance.Type, instance.AdditionalArguments); isInvalidAdditionalArguments {
+		allErrs = append(allErrs, field.Invalid(instancePath.Child("additionalArguments"), instance.AdditionalArguments,
+			fmt.Sprintf("Additional Arguments for %s are invalid! Valid values are: %s", instance.Type, reflect.ValueOf(allowedAdditionalArguments).MapKeys()),
+		))
+	}
+
 	databaselog.Info("Exiting instanceSpecValidatorForCreate...")
 	return allErrs
 }
 
+/* Checks if configured additional arguments are invalid */
+func isAdditionalArgumentsInvalid(dbType string, additionalArguments map[string]string) (bool, map[string]bool) {
+	allowedAdditionalArguments, err := util.GetAllowedAdditionalArgumentsForType(dbType)
+
+	if additionalArguments == nil {
+		return true, allowedAdditionalArguments
+	}
```

Comment:
I changed the logic of the function. It now returns an error instead of a bool, map tuple. This makes this less confusing and more concise. Also allows me to return descriptive error messages that are logged in the parent caller.